renderdoc/driver/vulkan/wrappers/vk_cmd_funcs.cpp

```diff
@@ -1780,10 +1763,10 @@ bool WrappedVulkan::Serialise_vkCmdBeginRenderPass(SerialiserType &ser, VkComman
         commandBuffer = RerecordCmdBuf(m_LastCmdBufferID);
 
         // only if we're partially recording do we update this state
-        if(ShouldUpdateRenderState(m_LastCmdBufferID, true))
+        if(IsCommandBufferPartialPrimary(m_LastCmdBufferID) || m_OutsideCmdBuffer != VK_NULL_HANDLE)
```

Comment:
I think there is still value in having a small utility function similar to `ShouldUpdateRenderState` to do this logic even if it's minimal, if only to avoid 'leaking' the awareness of `m_OutsideCmdBuffer` into the implementation of command replays like this.

Worst case it's a slightly unnecessary helper function, but in case the command buffer handling is refactored in future it's good to keep the exact replaying details more central.